fluent-react/src/localized.js

```diff
@@ -51,158 +51,134 @@ function toArguments(props) {
  *  translation is available.  It also makes it easy to grep for strings in the
  *  source code.
  */
-export default class Localized extends Component {
-  componentDidMount() {
-    const { l10n } = this.context;
-
-    if (l10n) {
-      l10n.subscribe(this);
-    }
+function Localized(props) {
+  const { id, attrs, children: child = null } = props;
+  const { l10n, parseMarkup } = useContext(FluentContext);
+
+  if (l10n === null) {
+    throw new Error(
+      "<Localized/> needs to have a <LocalizationProvider/> up in the tree"
+    );
   }
 
-  componentWillUnmount() {
-    const { l10n } = this.context;
-
-    if (l10n) {
-      l10n.unsubscribe(this);
-    }
+  // Validate that the child element isn't an array
+  if (Array.isArray(child)) {
+    throw new Error("<Localized/> expected to receive a single " +
+      "React node child");
   }
 
-  /*
-   * Rerender this component in a new language.
-   */
-  relocalize() {
-    // When the `ReactLocalization`'s fallback chain changes, update the
-    // component.
-    this.forceUpdate();
+  if (!l10n) {
```

Comment:
This line is never called, as it is checked on line 58 and throws exception

Reply:
In an ideal world I'd agree, but given this is JS and other things are falsy, I'm not sure if this holds up. @stasm might know.

Also in this PR I wasn't looking into changing Fluent's behavior but rather just move it to the new Context API (mostly because I'm not super knowledgable about the inner workings of Fluent)

Reply:
yeap, `null` it's not the same as falsy evaluation, but
```javascript
if (!l10n) { // if it was null it would return the wrapped component
   // Use the wrapped component as fallback.
   return child;
}
```
can't it simply return the child even if it's `null`?

Reply:
Maybe, I'm just not confident enough about the codebase to make this change. Once Stas is back, he might have an opinion on it